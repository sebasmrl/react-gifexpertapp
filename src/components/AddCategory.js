import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => { //aqui viene un objeto llamado props sino se desestructura despues se llamaria props.setCategories

    const [inputValue, setInputValue] = useState(''); //el input por defecto esta vacio por el valor inicial del state

    
    const handdleInputChange = ( e ) => {
        //console.log(e.target.value)
        //para que se pueda escribir en el input
        setInputValue(e.target.value) //esto el valor del input
        //console.log('handdleInputChange llamado');
    }

    const handdleSubmit = (e) => {
        e.preventDefault()
       // console.log('HanddleSubmit llamado'); 
        //console.log('Submit Hecho', )

        //validar qe}ue el texto sea lo sufientemente extenso
        if(inputValue.trim().length>2){
            setCategories(cats => [inputValue, ...cats]) //cats ->categories  -primero el evento iputChange y luego el enter y queda lo ultimo escrito
            setInputValue('');
        }
    } 

    //Debo retornar un jsx para que se pinte
    return (
            <form onSubmit={handdleSubmit}>
                <p>{inputValue}</p> {/* Solo es para las pruebas*/}
                {/* <h1>{inputValue}</h1> */}
                <input 
                    type='text'
                    value={inputValue}
                    onChange={handdleInputChange}
                />  
            </form>
    ) 
}

AddCategory.propTypes = {
        setCategories : PropTypes.func.isRequired
    }
