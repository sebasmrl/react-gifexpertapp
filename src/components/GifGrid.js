import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import {GifGridItem} from './GifGridItem'

export const GifGrid = ({category}) => {


    //Hooks - custom hooks
    //mando la categoria para hacer el  getGifs(category) dentro del customhook
    const {data: images, loading } = useFetchGifs(category); //images = [{id,title,url}, {id,title,url}]
    //console.log(loading)
    //un nuevo mensaje en GifGrid el 27/08/2021 a la 5:49pm
    
    return (
        <>
        <h3 className="animate__animated animate__fadeIn">{category}</h3>

        {/* si loading es true entonces evalua la segunda porque en AND hace eso, y en si termina mostrando el componente, depues cabian es estado y al volverse a renderizar loading vale false y no se pinta el parrafo ""loading*/}
        {loading && <p className="animate__animated animate__flash">Loading</p> }
        
        <div className="card-grid">
                
                {
                    images.map( img =>(
                         <GifGridItem 
                            key={img.id}
                            {...img}
                         />
                    ))
                }
                
        </div>
        </>
    )
}
