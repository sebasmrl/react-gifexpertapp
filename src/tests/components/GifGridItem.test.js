import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";



describe("Pruebas en <GifGridItem />", ()=>{
    const obj= {
        title: "Saitama", 
        url:"https://media3.giphy.com/media/DuVRadBbaX6A8/giphy.gif?cid=d90e7e9ayw79ydhltyyk5vvpy472z3czrl9ao2vk3j35tb4l&rid=giphy.gif&ct=g"
    };

    const wrapper = shallow(<GifGridItem { ...obj}/>);

    test('Se debe mostrar el componente correctamente', () => {
        
        
        
        expect(wrapper).toMatchSnapshot();
    });


    test('Debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim() ).toBe( obj.title);

    });

    test('Debe de tener la imagen igual al url y alt de los props', () => {
        const img =wrapper.find('img'); // com la imagen es una sola en componente colocamos el selctor en img pero podria ser por id #img-portada
       // console.log(img.html()); trabajar con base a un string
       //analizar con enzyme las propiedades (props) del componente <img> para
       //las props son src: "" y alt: ''
       //console.log(img.props().src);
       //console.log(img.prop('src')); //evaluar solo una propiedad 
       expect(img.prop('src')).toBe(obj.url);
       expect(img.prop('alt')).toBe(obj.title);
    });

    test('Debe tener animate__fadeIn', () => {
        const div = wrapper.find('div');

        expect(div.prop('className').includes('animate__fadeIn')).toBe(true); 
        //expect(div.prop('className').includes('animate__fadeIn')).not.toBe(false); //negacion
    })
    
    
    
    




})