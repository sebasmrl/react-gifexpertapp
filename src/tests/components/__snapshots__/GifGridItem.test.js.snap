// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Se debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Saitama"
    src="https://media3.giphy.com/media/DuVRadBbaX6A8/giphy.gif?cid=d90e7e9ayw79ydhltyyk5vvpy472z3czrl9ao2vk3j35tb4l&rid=giphy.gif&ct=g"
  />
  <p>
    Saitama
  </p>
</div>
`;
